src/mlpack/tests/main_tests/kernel_pca_test.cpp

```diff
@@ -0,0 +1,356 @@
+/**
+ * @file kernel_pca_test.cpp
+ * @author Saksham Bansal
+ *
+ * Test mlpackMain() of kernel_pca_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "KernelPrincipalComponentsAnalysis";
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/kernel_pca/kernel_pca_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct KernelPCATestFixture
+{
+ public:
+  KernelPCATestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~KernelPCATestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+static void ResetSettings()
+{
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+}
+
+BOOST_FIXTURE_TEST_SUITE(KernelPCAMainTest, KernelPCATestFixture);
+
+/**
+ * Make sure that all valid kernels return correct output dimension.
+ */
+BOOST_AUTO_TEST_CASE(KernelPCADimensionTest)
+{
+  std::string kernels[] = {
+      "linear", "gaussian", "polynomial",
+      "hyptan", "laplacian", "epanechnikov", "cosine"
+  };
+
+  for (std::string& kernel: kernels)
+  {
+    ResetSettings();
+    arma::mat x = arma::randu<arma::mat>(5, 5);
+    // Random input, new dimensionality of 3.
+    SetInputParam("input", std::move(x));
+    SetInputParam("new_dimensionality", (int) 3);
+    SetInputParam("kernel", kernel);
+    mlpackMain();
+
+    // Now check that the output has 3 dimensions.
+    BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_rows, 3);
+    BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_cols, 5);
+  }
+}
+
+/**
+ * Check that error is thrown when no kernel is specified.
+ */
+BOOST_AUTO_TEST_CASE(KernelPCANoKernelTest)
+{
+  arma::mat x = arma::randu<arma::mat>(5, 5);
+
+  SetInputParam("input", std::move(x));
+  SetInputParam("new_dimensionality", (int) 3);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Check that error is thrown when an invalid kernel is specified.
+ */
+BOOST_AUTO_TEST_CASE(KernelPCAInvalidKernelTest)
+{
+  arma::mat x = arma::randu<arma::mat>(5, 5);
+
+  SetInputParam("input", std::move(x));
+  SetInputParam("new_dimensionality", (int) 3);
+  SetInputParam("kernel", (std::string) "badName");
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Make sure if 0 dimensions is specified, we get a dataset with same
```

Comment:
I'd prefer if it were something like `Ensure for zero dimensionality, ...`

Reply:
Done during merge; 1171da89

Reply:
Thanks. I was traveling for a few days (without laptop) and couldn't make the changes.